src/main/java/org/truffleruby/core/regexp/ClassicRegexp.java

```diff
@@ -97,22 +97,25 @@ private static Regex makeRegexp(RubyContext runtime, RopeBuilder bytes, RegexpOp
         }
     }
 
-    static Regex getRegexpFromCache(RubyContext runtime, RopeBuilder bytes, Encoding enc, RegexpOptions options) {
-        if (runtime == null) {
-            Regex regex = makeRegexp(runtime, bytes, options, enc);
+    static Regex getRegexpFromCache(RubyContext context, RopeBuilder bytes, Encoding enc, RegexpOptions options) {
+        if (context == null) {
+            final Regex regex = makeRegexp(context, bytes, options, enc);
             regex.setUserObject(bytes);
             return regex;
         }
-        Rope key = RopeOperations.ropeFromRopeBuilder(bytes);
-        RopeKey ropeKey = new RopeKey(key, runtime.getHashing());
-        Regex regex = runtime.getRegexpCache().get(ropeKey);
+
+        final Rope key = RopeOperations.ropeFromRopeBuilder(bytes);
+        final RopeKey ropeKey = new RopeKey(key, context.getHashing());
+
+        final Regex regex = context.getRegexpCache().get(ropeKey);
         if (regex != null && regex.getEncoding() == enc && regex.getOptions() == options.toJoniOptions()) {
             return regex;
         }
-        regex = makeRegexp(runtime, bytes, options, enc);
-        regex.setUserObject(bytes);
-        runtime.getRegexpCache().put(ropeKey, regex);
-        return regex;
+
+        final Regex newRegex = makeRegexp(context, bytes, options, enc);
+        newRegex.setUserObject(bytes);
```

Comment:
I'm not sure where we use the "user object", if anywhere at all, but it might be a natural place to store the `ropeKey` if you're looking for ways to keep that live.

Reply:
We don't need to keep the key alive as it's a WeakValuedMap now :)